helpers/Physics.h

```diff
@@ -0,0 +1,320 @@
+						pPrevRot->y = angY;
+					}
+				}
+			} else {
+				if (strand->constraint.rotStepCalc.x) {
+					tempAngleX = RADF_TO_BINANG(jointTable[i - 2].rot.x);
+					workAngleX = RADF_TO_BINANG(-Math_Atan2F(sqrtf(SQ(workVec.x) + SQ(workVec.z)), workVec.y));
+					Math_SmoothStepToS(&tempAngleX, workAngleX, 1, DEGF_TO_BINANG(strand->constraint.rotStepCalc.x), 1);
+					angX = BINANG_TO_RAD(tempAngleX);
+					pPrevRot->x = angX;
+				}
+				if (strand->constraint.rotStepCalc.y) {
+					tempAngleY = RADF_TO_BINANG(jointTable[i - 2].rot.y);
+					workAngleY = RADF_TO_BINANG(Math_Atan2F(workVec.z, workVec.x));
+					Math_SmoothStepToS(&tempAngleY, workAngleY, 1, DEGF_TO_BINANG(strand->constraint.rotStepCalc.y), 1);
+					angY = BINANG_TO_RAD(tempAngleY);
+					pPrevRot->y = angY;
+				}
+			}
+		}
+		
+		Matrix_RotateY(angY, MTXMODE_NEW);
+		Matrix_RotateX(angX, MTXMODE_APPLY);
+		Matrix_MultZ(ABS(strand->limbsLength[i]) * strand->gfx.scale.z, &posAdd);
+		
+		// Pushes limbs away from selected Vec3f points
+		for (s32 i = 0; i < strand->spheres.num; i++) {
+			Vec3f tempPosAdd = {
+				pPrevPos->x + posAdd.x,
+				pPrevPos->y + posAdd.y,
+				pPrevPos->z + posAdd.z
+			};
+			f32 radiusTo = Math_Vec3f_DistXYZ(&tempPosAdd, &strand->spheres.centers[i]);
+			
+			if (radiusTo < strand->spheres.radius) {
+				s16 yaw = Math_Vec3f_Yaw(&strand->spheres.centers[i], &tempPosAdd);
+				posAdd.x += Math_SinS(yaw) * (strand->spheres.radius - radiusTo) * 0.5f;
+				posAdd.z += Math_CosS(yaw) * (strand->spheres.radius - radiusTo) * 0.5f;
+				velAdj.x += Math_SinS(yaw) * (strand->spheres.radius - radiusTo) * 0.5f;
+				velAdj.z += Math_CosS(yaw) * (strand->spheres.radius - radiusTo) * 0.5f;
+				
+				pPrevRot->y = angY =  Math_Atan2F(posAdd.z, posAdd.x);
+				pPrevRot->x = angX = -Math_Atan2F(sqrtf(SQ(posAdd.x) + SQ(posAdd.z)), posAdd.y);
+			}
+		}
+		
+		workVec.x = pPos->x;
+		workVec.y = pPos->y;
+		workVec.z = pPos->z;
+		
+		pPos->x = pPrevPos->x + posAdd.x;
+		pPos->y = pPrevPos->y + posAdd.y;
+		pPos->z = pPrevPos->z + posAdd.z;
+		
+		pVel->x = (pPos->x - workVec.x + velAdj.x) * strand->info.velMult;
+		pVel->y = (pPos->y - workVec.y + velAdj.y) * strand->info.velMult;
+		pVel->z = (pPos->z - workVec.z + velAdj.z) * strand->info.velMult;
+		
+		jointTable[i].vel = (Vec3f) {
+			CLAMP(pVel->x, -strand->info.maxVel, strand->info.maxVel),
+			CLAMP(pVel->y, -strand->info.maxVel, strand->info.maxVel),
+			CLAMP(pVel->z, -strand->info.maxVel, strand->info.maxVel),
+		};
+	}
+	
+	if (strand->gfx.noDraw) {
+		Matrix_Pull();
+		
+		return;
+	}
+	
+	Mtx* matrix = Graph_Alloc(gfxCtx, strand->info.numLimbs * sizeof(Mtx));
+	s16 y = RADF_TO_BINANG(jointTable[0].rot.y);
+	s16 x = RADF_TO_BINANG(jointTable[0].rot.x);
+	
+	for (i = 0; i < strand->info.numLimbs; i++) {
+		pPos = &jointTable[i].pos;
+		pRot = &jointTable[i].rot;
+		
+		Matrix_Translate(pPos->x, pPos->y, pPos->z, MTXMODE_NEW);
+		
+		if (strand->constraint.rotStepDraw.y) {
```

Comment:
```suggestion
		if (strand->constraint.rotStepDraw.y != 0) {
```
and for `rotStepDraw.x`